### DISCUSSION TITLE
How to not update some field if the value is null / '' / undefined ?

### DISCUSSION THREAD

**Discussion by mandaputtra** [prisma/prisma#5592] [Q&A] [answered]
If I run the query below it will update the password to `''`, how to not update the field if it is `'<empty-string>'` without needing to make another type of query?

```ts
    let password = '';
    if (updateUserDto.password) {
      password = await argon2.hash(updateUserDto.password);
    }
    return this.prisma.user.update({
      where: { id },
      data: {
        name: updateUserDto.name,
        email: updateUserDto.email,
        password,
        phoneNumber: updateUserDto.phoneNumber,
        role: updateUserDto.role,
      },
    });
```

**Comment by ryands17** [comment, marked as answer]
@mandaputtra 👋 

You can do this in the following manner:

```ts
 let password = '';
    if (updateUserDto.password) {
      password = await argon2.hash(updateUserDto.password);
    }
    return this.prisma.user.update({
      where: { id },
      data: {
        name: updateUserDto.name,
        email: updateUserDto.email,
      ...(password && { password }),
        phoneNumber: updateUserDto.phoneNumber,
        role: updateUserDto.role,
      },
});
```

**Reply by mandaputtra** [reply]
Hi @ryands17 :wave: 

I actually find another way of doing that, turns out if it undefined it won't update the field, so I do it like this

```
    let password: string;
    if (updateUserDto.password) {
      password = await argon2.hash(updateUserDto.password);
    }
    return this.prisma.user.update({
      where: { id },
      data: {
        name: updateUserDto.name,
        email: updateUserDto.email,
        password,
        phoneNumber: updateUserDto.phoneNumber,
        role: updateUserDto.role,
      },
    });

```

**Reply by mistersingh179** [reply]
yes undefined won't have an impact on the update, but null would. then depending on your schema this would either set to null or throw an error.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
